tests/integration/test_docker.py

```diff
@@ -415,4 +415,41 @@ def test_layers():
         reversed = image.layers(reversed=False)
         assert reversed[-1].inspect()['ContainerConfig']['Cmd'] == punchbag_cmd
 
-    
\ No newline at end of file
+
+def test_metadata():
+    with DockerBackend() as backend:
+        image = backend.ImageClass(FEDORA_MINIMAL_REPOSITORY, tag=FEDORA_MINIMAL_REPOSITORY_TAG)
+
+        c = image.run_via_binary(
+            DockerRunBuilder(command=["cat"], additional_opts=['-i',
+                                                               '-t',
+                                                               '--name', 'my_container',
+                                                               '-p', '1234:12345',
+                                                               '-p', '123:12345',
+                                                               '--hostname', 'my_hostname',
+                                                               '-e', 'ENV1=my_env',
+                                                               '-e', 'ASD=',
+                                                               '-e', 'A=B=C=D',
+                                                               '-e', 'XYZ',
+                                                               '-l', 'testlabel1=testvalue1'
+                                                               ])
+        )
+
+        container_metadata = c.get_container_metadata()
+
+        try:
+            assert container_metadata.command == ["cat"]
+            assert container_metadata.name == "my_container"
+            assert container_metadata.env_variables["ENV1"] == "my_env"
+            assert container_metadata.env_variables["ASD"] == ""
+            assert container_metadata.env_variables["A"] == "B=C=D"
+            assert container_metadata.hostname == "my_hostname"
+            assert "XYZ" not in list(container_metadata.env_variables.keys())
+            assert container_metadata.port_mappings["1234"] == "12345/tcp"
```

Comment:
`container_metadata.port_mappings` actually returns `{'12345/tcp': [1234, 123]}` in this case.